### DISCUSSION TITLE
Lot of COMMAND operation had invoked by Spring boot lettuce client to Redis cluster

### DISCUSSION THREAD

**Discussion by MyneniMahesh** [redis/lettuce#1792] [Q&A]
We are using lettuce redis client to connect to our redis cluster in production. We had a performance issue. while debugging we have identified that lot(almost 40K) COMMAND operation had invoked by few services. we couldn't able to reproduce the issue in local environment. Can some one guide us, which component is triggering the COMMAND operation. The use of COMMAND operation

**Comment by mp911de** [comment]
Lettuce issued in Redis Cluster mode a `COMMAND` command to identify which commands were available upon connecting the cluster. Sounds as if you're using multiple connections without connection pooling.

**Reply by MyneniMahesh** [reply]
@mp911de We are not using any connection polling. we are in a assumption that lettuce doesn't require any connection polling https://lettuce.io/core/release/reference/#connection-pooling.is-connection-pooling-necessary.

Can we exclude the invocation Command operation for the new connections ?
Do we have to configure the connection pooling in our case?

**Reply by venkatnpedada** [reply]
@mp911de Im in assumption that lettuce client doesn't need any connection pooling by default. could you please clarify why it need connection pooling? https://github.com/lettuce-io/lettuce-core/wiki/Connection-Pooling

**Reply by mp911de** [reply]
First, there's no general right or wrong. It always depends on connection usage and how you interact with Redis. Spring Data specifically uses a concept of connection sharing for some commands and for some commands (pipelining, BLPOP/BRPOP/XREAD with blocking) it uses dedicated connections to avoid connection blocking.

Allocating dedicated connections is a good use-case for pooling to buffer the expensive connection bootstrap.
